app/models/client.rb

```diff
@@ -0,0 +1,34 @@
+require './app/models/parser'
+require 'user_agent_parser'
+
+class Client < ActiveRecord::Base
+  include Parser
+
+  validates :identifier, :root_url, presence: true
+
+  has_many :payloads
+  has_many :referred_bies,  through: :payloads
+  has_many :request_types,  through: :payloads
+  has_many :event_names,    through: :payloads
+  has_many :u_agents,       through: :payloads
+  has_many :resolutions,    through: :payloads
+  has_many :ips,            through: :payloads
+  has_many :urls,           through: :payloads
+
+  def populate(payload)
```

Comment:
we talked about this during your review, but i want to mention it here. it's not a `Client`'s responsibility to parse & populate the `Payload` object. we should extract this logic out to separate module